packages/astro/src/core/build/plugin.ts

```diff
@@ -74,7 +69,7 @@ export function createPluginContainer(options: StaticBuildOptions, internals: Bu
 			const mutations = new Map<
 				string,
 				{
-					build: 'server' | 'client';
+					targets: BuildTarget[];
```

Comment:
Since we now have three build modes, the internal build plugin system has been refactored to accept an array of `'server' | 'content' | 'client'` rather than a single value.